Submodule boringssl/src/main/native/src contains modified content
diff --git a/boringssl/src/main/native/src/util/generate_build_files.py b/boringssl/src/main/native/src/util/generate_build_files.py
index 7f7232112..afa228b7a 100644
--- a/boringssl/src/main/native/src/util/generate_build_files.py
+++ b/boringssl/src/main/native/src/util/generate_build_files.py
@@ -90,12 +90,21 @@ class Android(object):
       self.PrintDefaults(blueprint, 'libpki_sources', files['pki'])
 
     # Legacy Android.mk format, only used by Trusty in new branches
-    with open('sources.mk', 'w+') as makefile:
+    with open('sources.mk', 'w+', newline='') as makefile:
       makefile.write(self.header)
       makefile.write('\n')
       self.PrintVariableSection(makefile, 'crypto_sources', files['crypto'])
       self.PrintVariableSection(makefile, 'crypto_sources_asm',
                                 files['crypto_asm'])
+      self.PrintVariableSection(makefile, 'crypto_sources_nasm',
+                                files['crypto_nasm'])
+      self.PrintVariableSection(makefile, 'ssl_sources', files['ssl'])
+      self.PrintVariableSection(makefile, 'tool_sources', files['tool'])
+      self.PrintVariableSection(makefile, 'test_support_sources',
+                                files['test_support'])
+      self.PrintVariableSection(makefile, 'crypto_test_sources',
+                                files['crypto_test'])
+      self.PrintVariableSection(makefile, 'ssl_test_sources', files['ssl_test'])
 
   def PrintDefaults(self, blueprint, name, files, asm_files=[], data=[]):
     """Print a cc_defaults section from a list of C files and optionally assembly outputs"""
